tests/lib/venv.py

```diff
@@ -4,30 +4,45 @@
 import sys
 import textwrap
 import venv as _venv
+from typing import TYPE_CHECKING, Optional
 
 import virtualenv as _virtualenv
 
 from .path import Path
 
+if TYPE_CHECKING:
+    # Literal was introduced in Python 3.8.
+    from typing import Literal
+
 
 class VirtualEnvironment:
     """
     An abstraction around virtual environments, currently it only uses
     virtualenv but in the future it could use pyvenv.
     """
 
-    def __init__(self, location, template=None, venv_type=None):
-        assert template is None or venv_type is None
-        assert venv_type in (None, "virtualenv", "venv")
+    def __init__(
+        self,
+        location: str,
+        template: Optional["VirtualEnvironment"] = None,
+        venv_type: 'Optional[Literal["virtualenv", "venv"]]' = None,
+    ):
         self.location = Path(location)
-        self._venv_type = venv_type or template._venv_type or "virtualenv"
+        assert template is None or venv_type is None
+        self._venv_type: str
```

Comment:
```suggestion
        self._venv_type: 'Literal["virtualenv", "venv"]'
```

(Some `assert`s may be needed to make this work.)